### PR TITLE
Set CONN_MAX_AGE to 60 seconds -- fixes #710

### LINKED ISSUE #710: Set CONN_MAX_AGE in DATABASE_URL
[CONN_MAX_AGE](https://docs.djangoproject.com/en/1.9/ref/databases/#persistent-connections) defaults to 0. I think it would be nice for us to be more opinionated about it, setting it to a non-zero value by default.

[StackOverflow](http://stackoverflow.com/questions/19937257/what-is-a-good-value-for-conn-max-age-in-django) suggests 60, [RevSys](http://www.revsys.com/blog/2015/may/06/django-performance-simple-things/) suggests 600 (10 min). I lean toward 600 since we're using allauth and authenticated users tend to stick around a site longer.

### PR DESCRIPTION
**Pull request by browniebroke**
Setting the value in the production config rather than in the environment variable as it seems clearer like this.

### CODE REVIEW

**Comment by sfdye** [comment]
Can you set it as an environment variable in .env? Which if the user didn't set, will default to 0. IMO, `CONN_MAX_AGE` does not really apply to local environment.

**Comment by browniebroke** [comment]
Not sure I understand this:

> IMO, CONN_MAX_AGE does not really apply to local environment.

Is it a comment on something I've done wrong? I've updated `production.py` config, which isn't used by the local environment, as far as I know, and `.env` file has `DJANGO_SETTINGS_MODULE=config.settings.production` in it.

**Comment by sfdye** [comment]
What I mean is have something like:

```python
# base.py

DATABASES['default']['CONN_MAX_AGE'] = env.int('CONN_MAX_AGE', default=0)
```

Users can then override this value on the `.env` file to whatever they want, you shouldn't just hard-code the value.

```bash
# .env file

CONN_MAX_AGE=60
```

**Comment by browniebroke** [comment]
I see, thanks! I've updated the PR.

**Comment by sfdye** [comment]
@webyneter @luzfcb what do you think?

File: {{cookiecutter.project_slug}}/config/settings/production.py [hunk 1]
```diff
@@ -182,12 +182,14 @@
...
         'HOST': env('RDS_HOSTNAME'),
         'PORT': env('RDS_PORT'),
+        'CONN_MAX_AGE': env.int('CONN_MAX_AGE', default=60),
     }
 }
 {% else %}
 # Use the Heroku-style specification
 # Raises ImproperlyConfigured exception if DATABASE_URL not in os.environ
 DATABASES['default'] = env.db('DATABASE_URL')
+DATABASES['default']['CONN_MAX_AGE'] = env.int('CONN_MAX_AGE', default=60)
```
Comment (webyneter) [inline {{cookiecutter.project_slug}}/config/settings/production.py]: The `default` argument should be refactored to avoid duplication, IMO.

Comment (webyneter) [reply {{cookiecutter.project_slug}}/config/settings/production.py]: Make it smth like `_DEFAULT_CONN_MAX_AGE` so the name indicates its intricate nature.

Comment (browniebroke) [reply {{cookiecutter.project_slug}}/config/settings/production.py]: Is there a way to set this globally without adding to the list of questions? I know of Jinja's `{% set ... %}` but that would still be duplicated between the 2 files `production.py` and `.env`.

Comment (webyneter) [reply {{cookiecutter.project_slug}}/config/settings/production.py]: @browniebroke I don't think we need to make this a cookiecutter option -- merely introducing `_DEFAULT_CONN_MAX_AGE = 60` to `production.py` should suffice; as for the `env.example`, I suggest that you leave its `CONN_MAX_AGE` blank (`CONN_MAX_AGE=`).

File: tests/test_cookiecutter_generation.py [hunk 2]
```diff
@@ -84,3 +84,14 @@ def test_flake8_compliance(cookies):
...
+
+def test_connection_max_age(cookies):
+    result = cookies.bake()
+
+    prod_config_file = result.project.join('config', 'settings', 'production.py')
+    prod_config_lines = prod_config_file.read()
+
+    expected = "DATABASES['default']['CONN_MAX_AGE'] = env.int('CONN_MAX_AGE', default=60)"
+
+    assert expected in prod_config_lines, prod_config_lines
```
Comment (webyneter) [inline tests/test_cookiecutter_generation.py]: Why the two of `prod_config_lines` in `assert expected in prod_config_lines, prod_config_lines`?

Comment (browniebroke) [reply tests/test_cookiecutter_generation.py]: Forgot to clean that up, the second one was to print the content of the file when the assertion was failing...

File: {{cookiecutter.project_slug}}/config/settings/production.py (hunk 1 above)
Comment (browniebroke) [reply {{cookiecutter.project_slug}}/config/settings/production.py]: Ok agreed, I will do that. Thanks for your help.

File: tests/test_cookiecutter_generation.py [hunk 3]
```diff
@@ -84,3 +84,14 @@ def test_flake8_compliance(cookies):
...
+
+def test_connection_max_age(cookies):
+    result = cookies.bake()
+
+    prod_config_file = result.project.join('config', 'settings', 'production.py')
+    prod_config_lines = prod_config_file.read()
+
+    expected = "DATABASES['default']['CONN_MAX_AGE'] = env.int('CONN_MAX_AGE', default=60)"
+
+    assert expected in prod_config_lines
```
Comment (webyneter) [inline tests/test_cookiecutter_generation.py]: @browniebroke I'm not sure what do we need this test for?..

Comment (browniebroke) [reply tests/test_cookiecutter_generation.py]: I've used it to verify that my template variable was rendered properly.

Comment (webyneter) [reply tests/test_cookiecutter_generation.py]: I have mixed feelings about that test: Should the default value change the test will also fail albeit altering the default value is a _non-functional_ change.

**Comment by browniebroke** [comment]
I don't have any strong opinions. Any thoughts from the other maintainers?
Who has the final word on this? I'll remove it if no one else object.

**Comment by webyneter** [comment]
@browniebroke please, leave a comment in the conversation, so the context of discussion is preserved.

File: tests/test_cookiecutter_generation.py (hunk 3 above)
Comment (browniebroke) [reply tests/test_cookiecutter_generation.py]: I don't have any strong opinions. Any thoughts from the other maintainers?
Who has the final word on this? I'll remove it if no one else object.

Comment (webyneter) [reply tests/test_cookiecutter_generation.py]: @pydanny, @jayfk, @luzfcb what do you think?

Comment (pydanny) [reply tests/test_cookiecutter_generation.py]: LGTM

**Pull request merged**